atlasdb-container-test-utils/src/main/java/com/palantir/atlasdb/containers/CassandraResource.java

```diff
@@ -16,19 +16,43 @@
 
 package com.palantir.atlasdb.containers;
 
+import java.util.Optional;
+import java.util.function.Supplier;
+
 import org.junit.rules.ExternalResource;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
 
 import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfig;
+import com.palantir.atlasdb.config.LeaderConfig;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
 import com.palantir.atlasdb.keyvalue.cassandra.CassandraKeyValueService;
 import com.palantir.atlasdb.keyvalue.cassandra.CassandraKeyValueServiceImpl;
+import com.palantir.atlasdb.keyvalue.impl.KvsManager;
+import com.palantir.atlasdb.keyvalue.impl.TestResourceManager;
+import com.palantir.atlasdb.keyvalue.impl.TmManager;
+import com.palantir.atlasdb.transaction.api.TransactionManager;
 
-public class CassandraResource extends ExternalResource {
+public class CassandraResource extends ExternalResource implements KvsManager, TmManager {
+    public static final Optional<LeaderConfig> LEADER_CONFIG = CassandraContainer.LEADER_CONFIG;
```

Comment:
why is this `Optional`? and why not just use `CassandraContainer.LEADER_CONFIG` instead of reexporting it here

Reply:
CassandraKeyValueServiceImpl constructor (and factories) take an Optional, so would otherwise have to wrap it on every call. Don't want users to have to use the one from CassandraContainer as it leaks abstraction and is harder to use

Reply:
Not sure I follow, it's just config no?